src/metaxy/metadata_store/postgres.py

```diff
@@ -0,0 +1,257 @@
+
+        Raises:
+            ValueError: If neither connection_string nor connection parameters provided.
+            ImportError: If Ibis or psycopg2 driver not installed.
+
+        Note:
+            When using SHA256 hash algorithm, pgcrypto extension is required.
+            The store attempts to enable it automatically on open unless
+            `enable_pgcrypto=False`.
+        """
+        params: dict[str, Any] = dict(connection_params or {})
+
+        explicit_params = {
+            "host": host,
+            "port": port,
+            "user": user,
+            "password": password,
+            "database": database,
+            "schema": schema,
+        }
+        for key, value in explicit_params.items():
+            if value is not None:
+                params.setdefault(key, value)
+
+        if connection_string is None and not params:
+            raise ValueError(
+                "Must provide either connection_string or connection parameters. "
+                "Example: connection_string='postgresql://user:pass@localhost:5432/db' "
+                "or host='localhost', database='db'."
+            )
+
+        if connection_string is None and "port" not in params:
+            params["port"] = 5432
+
+        self.host = params.get("host")
+        self.port = params.get("port")
+        self.database = params.get("database")
+        self.schema = params.get("schema")
+        self.enable_pgcrypto = enable_pgcrypto
+
+        super().__init__(
+            connection_string=connection_string,
+            backend="postgres" if connection_string is None else None,
+            connection_params=params if connection_string is None else params or None,
+            fallback_stores=fallback_stores,
+            **kwargs,
+        )
+
+    def _get_default_hash_algorithm(self) -> HashAlgorithm:
+        """Get default hash algorithm for PostgreSQL stores.
+
+        Uses MD5 as it's built-in and requires no extensions.
+        For SHA256 support, explicitly set hash_algorithm=HashAlgorithm.SHA256
+        and ensure pgcrypto extension is enabled.
+        """
+        return HashAlgorithm.MD5
+
+    def open(self) -> None:
+        """Open connection to PostgreSQL and enable required extensions.
+
+        If using SHA256 hash algorithm and enable_pgcrypto=True, attempts to
+        enable the pgcrypto extension. Logs a warning if extension cannot be enabled.
+
+        Raises:
+            ImportError: If psycopg2 driver not installed.
+            Various database errors: If connection fails.
+        """
+        super().open()
+
+        # Auto-enable pgcrypto if SHA256 is configured
+        if self.enable_pgcrypto and self.hash_algorithm == HashAlgorithm.SHA256:
+            self._ensure_pgcrypto_extension()
+
+    def _ensure_pgcrypto_extension(self) -> None:
+        """Ensure pgcrypto extension is enabled for SHA256 support.
+
+        Attempts to create the extension if it doesn't exist. Logs a warning
```

Comment:
I think that's ok, but we need to clarify that automatic installation is not for production